builtInFunctions/migrateDataTrie.go

```diff
@@ -64,9 +65,18 @@ func (mdt *migrateDataTrie) ProcessBuiltinFunction(
 		NewVersion:   core.AutoBalanceEnabled,
 		TrieMigrator: dtm,
 	}
-	err = acntDst.AccountDataHandler().MigrateDataTrieLeaves(argsMigrateDataTrie)
-	if err != nil {
-		return nil, err
+
+	shouldMigrateAcntDst := bytes.Equal(acntDst.AddressBytes(), vmcommon.SystemAccountAddress) || !vmcommon.IsSystemAccountAddress(acntDst.AddressBytes())
+	if shouldMigrateAcntDst {
```

Comment:
why this complications ? why not !vmcommon.IsSystemAccountAddress(vmInput.RecvAddress)  - migrate normal account, otherwise migrate system account.